### PR TITLE
Corrected and updated developer docs

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
@davidhutchens can you please take a look?

### CODE REVIEW

**Review by davidhutchens — CHANGES_REQUESTED** [review]
Here are my suggestions.

File: docs/developers.md [hunk 1]
```diff
@@ -132,7 +132,10 @@ and build `Logisim-evolution` from source as described above.
 If you want to contribute to Logisim-evolution, this is how to do it:
 
 * Make a local *fork* of `Logisim-evolution` by clicking the *Fork* button on the
-  [project GitHub page](https://github.com/logisim-evolution/logisim-evolution).
+  [project GitHub page](https://github.com/logisim-evolution/logisim-evolution). This will create
+  copy of `Logisim-evolution` repository on your own Github account.
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/developers.md:136]: Done.

File: docs/developers.md [hunk 2]
```diff
@@ -132,7 +132,10 @@ and build `Logisim-evolution` from source as described above.
...
+* As all the development happens on [`develop` branch](https://github.com/logisim-evolution/logisim-evolution/tree/develop),
+  ensure you checkout it before you create own branch
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/developers.md:138]: Done

File: docs/style.md [hunk 3]
```diff
@@ -31,19 +32,18 @@ config file.
...
 
 Edit the existing Google Checks configuration:
 
 * Open the `Tools -> Checkstyle` plugin settings.
 * Activate the `Google Checks` configuration.
-* Highlight the `Google Checks` row and click the `Pen` icon above the list to edit it.
-* Look for the `org.checkstyle.google.suppressionfilter.config` named property and set its value
-  to `config/checkstyle/suppressions.xml`.
-* Click `Finish`.
+* Some rules should be disabled, so we also ship `checkstyle-suppressions.xml` config file. It
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/style.md:44]: Fixed.

File: docs/style.md [hunk 4]
```diff
@@ -31,19 +32,18 @@ config file.
...
+  lives in project's root directory and should be picked automatically and there should not be
+  necessary to do anything extra to make it used.
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/style.md:46]: Fixed.

File: docs/style.md [hunk 5]
```diff
@@ -53,17 +53,20 @@ You can configure InteliJ's CheckStyle plugin to behave exactly as we configure
...
-  * Click `Browse` and point to the `config/checkstyle/google_checks.xml` file.
+  * Click `Browse` and point to the `google_checks.xml` file.
   * Enable `Store relative to project location` and click `Next`.
-  * On the `Property` table, look for a property named `org.checkstyle.google.suppressionfilter.config`. Set the value
-    to `config/checkstyle/suppressions.xml` and click `Next.
+  * Some rules should be disabled, so we also ship `checkstyle-suppressions.xml` config file. It
+    lives in project's root directory and should be picked automatically and there should not be
+    necessary to do anything extra to make it used. If for any reason you need that changed, 
+    edit configuration and look for a property named `org.checkstyle.google.suppressionfilter.config`,
+    then set its value `checksyle-suppressions.xml` and click `Next.
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/style.md:69]: Fixed.

File: docs/style.md [hunk 6]
```diff
@@ -53,17 +53,20 @@ You can configure InteliJ's CheckStyle plugin to behave exactly as we configure
...
   * Set the description to `Logisim-evolution`.
   * Select `Use a local Checkstyle file`.
-  * Click `Browse` and point to the `config/checkstyle/google_checks.xml` file.
+  * Click `Browse` and point to the `google_checks.xml` file.
   * Enable `Store relative to project location` and click `Next`.
-  * On the `Property` table, look for a property named `org.checkstyle.google.suppressionfilter.config`. Set the value
-    to `config/checkstyle/suppressions.xml` and click `Next.
+  * Some rules should be disabled, so we also ship `checkstyle-suppressions.xml` config file. It
+    lives in project's root directory and should be picked automatically and there should not be
+    necessary to do anything extra to make it used. If for any reason you need that changed,
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply docs/style.md:67]: Done.

**Comment by MarcinOrlowski** [comment]
Thanks @davidhutchens.  Corrected all issues you spotted. PTAL.

**Review by davidhutchens — APPROVED** [review]
This looks good to me.

**Review by maehne — APPROVED** [review]
Looks good to me. At some point, we may consider to document how to use Checkstyle also in other IDEs, e.g., VS Code.

**Comment by maehne** [comment]
It might be also worth mentioning that it is common to create a branch for a topical fix/feature, which you then use to create the PR onto `develop`. Thus, the fix/feature can be easily evolved based on the review while also easing rebasing the branch or merging in changes to `develop` to keep in sync with upstream. Thus, a contributor can work easily on different topics in parallel without needing to wait that his PR gets merged upstream.

**Pull request merged**